lib/user_interface.rb

```diff
@@ -70,8 +58,24 @@ def display_contact(contact)
     end
   end
 
+  def add_another_contact?
+    output.print ANOTHER_CONTACT_PROMPT
+    collect_vaild_input { |user_input| user_input.match?(/[yn]/i) }.match?(/y/i)
+  end
+
   private
 
+  def collect_vaild_input
+    user_input = ''
+    loop do
+      user_input = input.gets.chomp
+      break if yield user_input
```

Comment:
I believe if you do `break user_input if yield(user_input)` then you don't need to initialize the variable before entering the loop and you also don't have to return it at the end because it would be returned there :+1: